pkg/certificates/renewerbottlerocket.go

```diff
@@ -0,0 +1,319 @@
+package certificates
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/aws/eks-anywhere/pkg/logger"
+)
+
+const (
+	brEtcdCertDir           = "/var/lib/etcd"
+	brControlPlaneCertDir   = "/var/lib/kubeadm/pki"
+	brControlPlaneManifests = "/var/lib/kubeadm/manifests"
+	brTempDir               = "/run/host-containerd/io.containerd.runtime.v2.task/default/admin/rootfs/tmp"
+)
+
+// BottlerocketRenewer implements OSRenewer for Bottlerocket systems.
+type BottlerocketRenewer struct {
+	osType OSType
+	backup string
+}
+
+// NewBottlerocketRenewer creates a new BottlerocketRenewer.
+func NewBottlerocketRenewer(backupDir string) *BottlerocketRenewer {
+	return &BottlerocketRenewer{
+		osType: OSTypeBottlerocket,
+		backup: backupDir,
+	}
+}
+
+// RenewControlPlaneCerts renews certificates for control plane nodes.
+func (b *BottlerocketRenewer) RenewControlPlaneCerts(
+	ctx context.Context,
+	node string,
+	cfg *RenewalConfig,
+	component string,
+	ssh SSHRunner,
+) error {
+	logger.V(0).Info("Renewing control-plane certificates", "node", node)
+
+	hasExternalEtcd := cfg != nil && len(cfg.Etcd.Nodes) > 0
+
+	if hasExternalEtcd {
+		if err := b.transferCertsToControlPlane(ctx, node, ssh); err != nil {
+			return fmt.Errorf("transferring certificates to control plane node: %v", err)
+		}
+	}
+
+	sessionCmds := b.sheltie(
+		b.pullContainerImage(),
+		b.backupControlPlaneCerts(component, hasExternalEtcd, b.backup, brControlPlaneCertDir),
+		b.renewControlPlaneCerts(),
+		b.checkControlPlaneCerts(),
+		b.copyExternalEtcdCerts(),
+		b.restartControlPlaneStaticPods(),
+	)
+
+	if _, err := ssh.RunCommand(ctx, node, sessionCmds); err != nil {
+		return fmt.Errorf("renewing control plane certificates: %v", err)
+	}
+
+	logger.V(0).Info("Renewed control-plane certificates", "node", node)
+	return nil
+}
+
+func (b *BottlerocketRenewer) transferCertsToControlPlane(
```

Comment:
SInce we have this for both ubuntu and bottlerocket, can we define this method in `OSRenewer` interface?